contracts/ModuleRegistry.sol

```diff
@@ -129,12 +130,16 @@ contract ModuleRegistry is IModuleRegistry, EternalStorage {
         } else {
             require(getBoolValue(Encoder.getKey("verified", _moduleFactory)), "ModuleFactory must be verified");
         }
-        require(isCompatibleModule(_moduleFactory, msg.sender), "Incompatible versions");
         // This if statement is required to be able to add modules from the STFactory contract during deployment
         // before the token has been registered to the STR.
-        if ((!_isUpgrade) && ISecurityTokenRegistry(getAddressValue(Encoder.getKey("securityTokenRegistry"))).isSecurityToken(msg.sender)) {
-            pushArray(Encoder.getKey("reputation", _moduleFactory), msg.sender);
-            emit ModuleUsed(_moduleFactory, msg.sender);
+        if (ISecurityTokenRegistry(getAddressValue(Encoder.getKey("securityTokenRegistry"))).isSecurityToken(msg.sender)
```

Comment:
To reduce gas, we can save STR address locally and then reuse it.

Reply:
This is only referenced in one place in this function now.